src/lib/dump_dir.c

```diff
@@ -755,10 +758,11 @@ uid_t dd_get_owner(struct dump_dir *dd)
 
     if (ret < 0)
     {
-        if (ret != -ENOENT)
-            return ret;
+        if (errorno && ret != -ENOENT)
+            *errorno = ret;
```

Comment:
I’m wondering… What do you think about simply using `errno`? There’s nothing that prevents us from doing so and documenting the magic `(uid_t) -1) value to mean “check errno”. 

Realistically, you would only reach the millions in the container world, but I see that at least my Fedora installation limits the subordinate IDs to 600100000 (which is well below `UINT32_MAX` and `INT32_MAX`).

Additional note is that the signedness of `uid_t` is undefined: https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_types.h.html, which would explain `MAX_UID_T`.